python/ray/train/v2/_internal/state/state_actor.py

```diff
@@ -20,18 +24,76 @@ def __init__(self):
         # {run_id: {attempt_id: TrainRunAttempt}}
         self._run_attempts: Dict[str, Dict[str, TrainRunAttempt]] = defaultdict(dict)
 
+        self._export_logger = self._init_export_logger()
+
     def create_or_update_train_run(self, run: TrainRun) -> None:
         self._runs[run.id] = run
+        self._maybe_export_train_run(run)
 
     def create_or_update_train_run_attempt(self, run_attempt: TrainRunAttempt):
         self._run_attempts[run_attempt.run_id][run_attempt.attempt_id] = run_attempt
+        self._maybe_export_train_run_attempt(run_attempt)
 
     def get_train_runs(self) -> Dict[str, TrainRun]:
         return self._runs
 
     def get_train_run_attempts(self) -> Dict[str, Dict[str, TrainRunAttempt]]:
         return self._run_attempts
 
+    # ============================
+    # Export API
+    # ============================
+
+    def is_export_api_enabled(self) -> bool:
+        return self._export_logger is not None
+
+    def _init_export_logger(self) -> Optional[logging.Logger]:
+        # Proto schemas should be imported within the scope of TrainStateActor to
+        # prevent serialization errors.
+        from ray.core.generated.export_event_pb2 import ExportEvent
+
+        # TODO: Should this be EventLogType.TRAIN_STATE?
+        export_api_enabled = check_export_api_enabled(
+            ExportEvent.SourceType.EXPORT_TRAIN_RUN
+        )
```

Comment:
@nikitavemuri wasn't sure what type of compatibility is needed for this, in particular for the existing submission job path.

Reply:
Let's keep the env var config for the event sources (for more granularity to disable for perf reasons) and not the log file names